### ISSUE TITLE
 Support for IOT Rule Action "errorAction"

### ISSUE THREAD

**Issue by legokichi** [terraform-providers/terraform-provider-aws#7147]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

We have been able to use error action in topic rule since 2017-12.

### New or Affected Resource(s)

* aws_iot_topic_rule

### Potential Terraform Configuration

```hcl
resource "aws_iot_topic_rule" "rule" {
    name = "${local.rule}"
    enabled = true
    sql = "SELECT { \"DeviceId\": topic(2), \"Timestamp\": Timestamp, \"payload\": payload, \"ttl\": cast(round(timestamp() / 1000 + 360) as String), } FROM 'dev/+/log"
    sql_version = "2016-03-23"
    dynamodb {
        role_arn = "${aws_iam_role.main.arn}"
        table_name = "${aws_dynamodb_table.main.name}"
        hash_key_field = "${aws_dynamodb_table.main.hash_key}"
        hash_key_value = "$${DeviceId}"
        range_key_field = "${aws_dynamodb_table.main.range_key}"
        range_key_value = "$${Timestamp}"
        payload_field = "payload"
    }
    error_action {
        dynamodb {
          role_arn = "${aws_iam_role.main.arn}"
          table_name = "${aws_dynamodb_table.main.name}"
          hash_key_field = "${aws_dynamodb_table.main.hash_key}"
          hash_key_value = "${local.rule}.error",
          range_key_field = "${aws_dynamodb_table.main.range_key}"
          range_key_value = "${timestamp()}"
          payload_field = "payload"
        }
    }
}

```

### References

* https://github.com/terraform-providers/terraform-provider-aws/issues/4534
* https://www.terraform.io/docs/providers/aws/r/iot_topic_rule.html
* https://aws.amazon.com/about-aws/whats-new/2017/12/amazon-iot-rules-engine-now-supports-error-actions/
* https://docs.aws.amazon.com/iot/latest/apireference/API_CreateTopicRule.html
* https://docs.aws.amazon.com/iot/latest/apireference/API_TopicRulePayload.html#iot-Type-TopicRulePayload-errorAction

**Comment by smougenot** [comment]
PR looks dormant for months :pray: help it

**Comment by bflad** [comment]
Support for the `error_action` configuration block (which itself supports any of the various "action" configuration blocks such as `dynamodb`) has been merged and will release with version 2.68.0 of the Terraform AWS Provider, later this week. Thanks to @kurtmc for the implementation. 👍
